### PR TITLE
Display upcoming and past events

### PR DESCRIPTION
**Pull request by flora8984461**
I am creating a separate branch since I want to compare with #109 
(Cuz for this branch I am having an annoying warning, the functions are all working fine, but I still wanna see if I can fix it...)

---------Update-2020-10-01-----------------------

- [x] Checkbox "Include past events"-By default, show upcoming events, if include past events, after upcoming events finished loading, show past events
- [x]  greyscale past events, and also include a header as a separation that says: Past Events before we list all the past events.
- [x]  remove the Unknown "Event type"

NOTE: 
- [x]  Fix the warning 
`Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.`

Signed-off-by: Yi Liu <yi.liu@eclipse-foundation.org>

### CODE REVIEW

**Comment by chrisguindon** [comment]
Can we show 20 events at a time?

**Comment by flora8984461** [comment]
> Can we show 20 events at a time?

Since I am handling upcoming and past events separately, is it okay that for each of them I set 20 one time? Like the current preview?

**Comment by chrisguindon** [comment]
@flora8984461 I would like to finish this task. Can you fix the conflicts and let me know if you are ready for review?

**Comment by flora8984461** [comment]
I think for now I am ready, but please check if all behave as expected. 

Thanks!

**Comment by flora8984461** [comment]
- [ ] 1) resolve this slidebar

image

- [ ] 2) remove "Sorry, we reached the end with this search" of upcoming events when showing past events

File: js/components/pastAndUpcomingEvents/EventLists.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <> 
+      {events.map((event) => (
+        <div className={`col-md-12 max-min-width event-${eventTime}`} key={event.id}>
+          <EventCard event={event} />
+        </div>
+      ))}
+
+      <div className="event-load-more">
+        { reachEnd ?
+        ( eventTime === "upcoming" && showPastEvents ? null : <p className="margin-5">Sorry, we reached the end with this search</p> ) :
```
Comment (autumnfound) [inline js/components/pastAndUpcomingEvents/EventLists.js:22]: We'll want to update the wording on this as it is syntactically odd. Try something along the lines of

> We have reached the end of results for the current filters

or

> No more results to show for current filters

Comment (flora8984461) [reply js/components/pastAndUpcomingEvents/EventLists.js]: Updated the text

**Comment by ericpoirier** [comment]
I'm reviewing this PR and I'm just wondering if there is something missing before we merge? I've been working on a patch for the Events API, are we waiting for that to be merged before merging this PR?

Thanks.

**Comment by chrisguindon** [comment]
> I'm reviewing this PR and I'm just wondering if there is something missing before we merge? I've been working on a patch for the Events API, are we waiting for that to be merged before merging this PR?

The API does not seems to be ready:
https://deploy-preview-111--eclipsefdn-events.netlify.app/

**Comment by ericpoirier** [comment]
> > I'm reviewing this PR and I'm just wondering if there is something missing before we merge? I've been working on a patch for the Events API, are we waiting for that to be merged before merging this PR?
> 
> The API does not seems to be ready:
> https://deploy-preview-111--eclipsefdn-events.netlify.app/

I'm still unable to rebuild staging or production because of permissions issues.

Once the site has been rebuilt, I'll report back here.

**Comment by chrisguindon** [comment]
@ericpoirier Thanks! The PR should be ready to be merge once the API is ready.

The next step would be to confirm with Shanda that all is well. PLease note that I don't expect you to merge this PR as we need the marketing team to test that all is well first.

**Comment by ericpoirier** [comment]
> @ericpoirier Thanks! The PR should be ready to be merge once the API is ready.
> 
> The next step would be to confirm with Shanda that all is well. PLease note that I don't expect you to merge this PR as we need the marketing team to test that all is well first.

The API has been successfully updated.

From what I can see, the changes that I did fixed both of these urls.

https://newsroom.eclipse.org/api/events?options[orderby][field_event_date]=custom

https://newsroom.eclipse.org/api/events?parameters[upcoming_only]=1&options[orderby][field_event_date]=ASC

**Comment by ericpoirier** [comment]
@shanda-eclipse can you have a look at the following url to make sure all the events are loading properly

https://deploy-preview-111--eclipsefdn-events.netlify.app/

Thanks!

**Comment by shanda-eclipse** [comment]
It does seem strange that when scrolling through the list, it moves directly from upcoming events to past events without any sort of indication. 

Hudson and I were also discussing the event types. I think we would have to do some clean up in the CMS to fix these. @hudsonkelly83 can you take a look?

**Comment by chrisguindon** [comment]
> It does seem strange that when scrolling through the list, it moves directly from upcoming events to past events without any sort of indication.

I expected that feedback :) We might need to wait for my next hire before we have time to finish this work.

> Hudson and I were also discussing the event types. I think we would have to do some clean up in the CMS to fix these. @hudsonkelly83 can you take a look?

I would recommend opening a separate issue for that. I would consider that outside of the scope of this issue.

**Pull request closed without merging**